src/Raven.Server/Documents/Handlers/QueriesHandler.cs

```diff
@@ -43,17 +44,41 @@ public Task Get()
         public async Task HandleQuery(HttpMethod httpMethod)
         {
             using (var tracker = new RequestTimeTracker(HttpContext, Logger, Database, "Query"))
-            using (var token = CreateTimeLimitedQueryToken())
-            using (Database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
             {
-                var debug = GetStringQueryString("debug", required: false);
-                if (string.IsNullOrWhiteSpace(debug) == false)
+                try
                 {
-                    await Debug(context, debug, token, tracker, httpMethod);
-                    return;
+                    using (var token = CreateTimeLimitedQueryToken())
+                    using (Database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                    {
+                        var debug = GetStringQueryString("debug", required: false);
+                        if (string.IsNullOrWhiteSpace(debug) == false)
+                        {
+                            await Debug(context, debug, token, tracker, httpMethod);
+                            return;
+                        }
+
+                        await Query(context, token, tracker, httpMethod);
+                    }
+                }
+                catch (Exception e)
+                {
+                    if (tracker.Query == null)
+                    {
+                        string errorMessage;
+                        if (e is EndOfStreamException || e is ArgumentException)
+                        {
+                            errorMessage = "Failed: " + e.Message;
+                        }
+                        else
+                        {
+                            errorMessage = "Failed: " +
+                                           HttpContext.Request.Path.Value +
+                                           e.ToString();
+                        }
+                        tracker.Query = errorMessage;
+                    }
+                    throw;
                 }
-
```

Comment:
What happens if the `tracker.Query` _isn't_ null? How does the user get the error?

Reply:
Its handled inside await Query(context, token, tracker, httpMethod); and the query is passed to the tracker.
There is a problem with case that if we didnt succeed to parse the query to memory OR didnt succeed to AllocateOperationContext then the tracker.Query remains null and this aims to change that.